stable/graylog/values.yaml

```diff
@@ -23,6 +23,14 @@ tags:
   install-mongodb: true
 
 graylog:
+
+  ## NOTE: This will NOT change the UID:GID used by the main container and affects ONLY initContainers created by this chart.
+  ## The following is represents the default UID:GID an initContainer should use if it decides to chown.
+  ## If UID:GID is changed future graylog images, this can be corrected here as well.
+  initContainers:
+    graylogUID: 1100
+    graylogGID: 1100
```

Comment:
I believe it is a pretty common convention to have something like this in `values.yaml`:

```yaml
securityContext:
  enabled: false
  fsGroup: 1100
  runAsUser: 1100
```

Then in your pod template:

```yaml
{{- if .Values.securityContext.enabled }}
securityContext:
  fsGroup: {{ .Values.securityContext.fsGroup }}
  runAsUser: {{ .Values.securityContext.runAsUser }}
{{- end }}
```

Any reason not to set securityContext on the statefulset? It seems like this would avoid drift between expected and actual user/group IDs.

Reply:
`securityContext` changes the user that runs inside the container. But that's not the cause of problem.

The graylog container itself already runs with `1100:1100`, and that's why it fails to access a `journal` directory owned by another `uid:gid`. If the `initContainer` from this chart gets the same uid:gid, it won't be able to `chown` the journal directory. It won't be able to do that with the same user as the graylog container.

I suppose the names used in `values.yaml`. My intention was to make it clear that we are not running the `initContainer` with a different uid:gid, but rather using those values to fix permissions.

Reply:
> `securityContext` changes the user that runs inside the container. But that's not the cause of problem.

Right, what I'm wondering though is whether there would be a reason _not_ to also make uid/gid configurable for the graylog container as part of this PR. If the graylog image switches to a different uid/gid (or if we wanted to make this configurable), then following convention now would prevent breaking changes in the future.

Reply:
That is a good point.

However, the graylog image creates its user at build ([Dockerfile](https://github.com/Graylog2/graylog-docker/blob/3.0.0-rc.1-2/Dockerfile)). I doubt this can be changed at runtime without side effects. That is a more complicated PR on their side.

This is not a critical issue, and it's particular to some use cases only. It can be troubling to workaround, but it's not impossible.

Maybe better comments on `values.yaml` would avoid confusion. But I'm open to putting this PR on hold until we find a better way to implement this.

Reply:
Cross relating this concern with the Graylog repo, [issue #76 on their side](https://github.com/Graylog2/graylog-docker/issues/76).

I'm still not sure about the proper way to resolve this.

Reply:
> Maybe better comments on `values.yaml` would avoid confusion. But I'm open to putting this PR on hold until we find a better way to implement this.

If this is currently preventing usage of the chart, then how about hardcoding the uid/gid in the init container for now (no changes to `values.yaml`) and then circling back with graylog folks to come up with a solution that is more extensible?

Reply:
Sounds good to me. I should be able to commit this later tonight.

Thank you, Jacob!

Reply:
If [PR #78](https://github.com/Graylog2/graylog-docker/pull/78) gets approved on their side, this one becomes unecessary. Let's see how that goes.

Reply:
@jlegrone 

Would it make sense to `chmod 777 /usr/share/graylog/data/journal`? That would make the journal directory writable to anyone, and there would no be a need to hardcode the uid:gid in the chart template. Keep in mind that could `chmod` only the root journal directory, and not its data and subdirectories.

Reply:
@juliohm1978 's chmod 777 might sound wrong from a security perspective, but since no one aside the system admins has access to the journal volume and also thinking the journal data is ephemeral I think it is a good solution.

Reply:
I was able to rebase the PR with the `chmod` solution. Let me know if anything else is required.